src/main/java/serenitylabs/tutorials/vetclinic/Pet.java

```diff
@@ -19,21 +27,53 @@ public Breed getBreed() {
         return breed;
     }
 
+    public double getWeightInKilos() {
+        return weightInKilos;
+    }
+
     public static PetBuilder dog() { return new PetBuilder(Breed.Dog);}
     public static PetBuilder cat() { return new PetBuilder(Breed.Cat);}
-    public static PetBuilder rabbit() { return new PetBuilder(Breed.Rabbit);}
-    public static PetBuilder parrot() { return new PetBuilder(Breed.Parrot);}
-    public static PetBuilder fish() { return new PetBuilder(Breed.Fish);}
+
+    public boolean isWellFed() {
+
+        double totalEaten = 0.0;
+
+        for(Meal meal : mealsGiven) {
+            if ((breed == Breed.Cat) && (meal.getFoodBrand() == PetFood.KittyKat)) {
+                totalEaten = totalEaten + meal.getAmountInGrams();
+            } else if ((breed == Breed.Dog) && (meal.getFoodBrand() == PetFood.FidosFood)) {
+                totalEaten = totalEaten + meal.getAmountInGrams();
+            }
+        }
+        double amountNeeded = 0.0;
+        if (breed == Breed.Cat) {
+            amountNeeded = getWeightInKilos() * 10;
+        } else if (breed == Breed.Dog) {
+            amountNeeded = getWeightInKilos() * 20;
+        }
```

Comment:
Here you are including logic about how much food is needed for each type of pet inside the Pet class. This means that the Pet class needs to know about all the possible business logic for different types of animals. It would be better to extract this knowledge into another class or classes that specialises in knowing how much a particular breed of pet needs to eat.